.github/workflows/pulsar-ci.yaml

```
@@ -498,6 +499,66 @@ jobs:
       - name: Delete docker image from GitHub Actions Artifacts
         run: |
           gh-actions-artifact-client.js delete pulsar-java-test-image.zst
+          
+
+  cpp-tests:
+    name: "CI - CPP, Python Tests"
+    runs-on: ubuntu-20.04
+    timeout-minutes: 120
+    if: ${{ steps.check_changes.outputs.docs_only != 'true' }}
+    needs: 'integration-tests'
+    steps:
+      - name: checkout
+        uses: actions/checkout@v2
+
+      - name: Tune Runner VM
+        uses: ./.github/actions/tune-runner-vm
+
+      - name: Cache local Maven repository
+        uses: actions/cache@v2
+        with:
+          path: |
+            ~/.m2/repository/*/*/*
+            !~/.m2/repository/org/apache/pulsar
+          key: ${{ runner.os }}-m2-dependencies-core-modules-${{ hashFiles('**/pom.xml') }}
+          restore-keys: |
+            ${{ runner.os }}-m2-dependencies-core-modules-
+
+      - name: Set up JDK 17
+        uses: actions/setup-java@v2
+        with:
+          distribution: 'temurin'
+          java-version: 17
+
+      - name: clean disk
+        run: |
+          sudo apt clean
+          docker rmi $(docker images -q) -f
+          df -h
+
+      - name: Install gh-actions-artifact-client.js
+        uses: apache/pulsar-test-infra/gh-actions-artifact-client/dist@master
+
+      - name: Restore maven build results from Github artifact cache
+        run: |
+          cd $HOME
+          $GITHUB_WORKSPACE/build/pulsar_ci_tool.sh restore_tar_from_github_actions_artifacts pulsar-maven-repository-binaries
+
+      - name: build cpp artifacts
+        run: |
+          echo "Build C++ client library"
+          pulsar-client-cpp/docker-build.sh
+
+      - name: run c++ tests
+        run: pulsar-client-cpp/docker-tests.sh
+
+      - name: Upload test-logs
+        uses: actions/upload-artifact@v3
+        if: failure()
+        continue-on-error: true
+        with:
+          name: cpp-test-logs
+          path: cpp-test-logs
```

```suggestion
          name: cpp-tests-logs
          path: test-logs
```
the name is for the artifact (used `cpp-tests` (id of the build job, could be something else unique too) + `-log`, the path is the directory which contains the logs and that's `test-logs`.